src/vm/database/marf.rs

```diff
@@ -148,47 +206,130 @@ impl <S> KeyValueStorage for &mut MarfedKV <S> where S: KeyValueStorage {
         result
     } 
 
-    fn put(&mut self, key: &str, value: &str) {
-        let marf_value = MARFValue::from_value(value);
+    pub fn put(&mut self, key: &str, value: &str) {
+        match self.marf {
+            Some(ref mut marf) => {
+                let marf_value = MARFValue::from_value(value);
 
-        self.side_store.put(&marf_value.to_hex(), value);
+                self.side_store.put(&marf_value.to_hex(), value);
 
-        self.marf.insert(key, marf_value)
-            .expect("ERROR: Unexpected MARF Failure")
+                marf.insert(key, marf_value)
+                    .expect("ERROR: Unexpected MARF Failure")
+            },
+            None => {
```

Comment:
Is this path ever taken outside of testing?  I saw the comment earlier about MARF-less data stores being used for raw evals, and while that's used in the CLI, I'm not sure that's ever used by the peer?  My concern here is that if we expect the peer to never take this code path, then this code path should panic when used by the peer to process a transaction.

Reply:
Right -- it's only ever used for testing and the raw evals.

I'll turn the MarfedKV into a trait and have two implementors, rather than use the option type. One would be used by the raw eval (and any other marf-less contexts we may have in the future, e.g. a scripting engine) and in some Clarity unit tests. The other will be the only one used by peers.

Reply:
Addressed in 84dbe6f